Aplib.Core/ILoggable.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// An interface that allows defining the structure loggable objects through a tree.
+    /// </summary>
+    public interface ILoggable : IDocumented
+    {
+        /// <summary>
+        /// Gets the children of the loggable object.
+        /// </summary>
+        /// <returns>The children of the loggable object.</returns>
+        public IEnumerable<ILoggable> GetChildren();
+
+        /// <summary>
+        /// Generates a log tree of the loggable object.
+        /// </summary>
+        /// <param name="depth">The depth of the first node of log tree.</param>
+        /// <returns>The first node of the log tree.</returns>
+        public LogNode GetLogTree(int depth = 0)
+        {
+            LogNode root = new(this, depth);
+            foreach (ILoggable child in GetChildren())
+            {
+                root.Children.Add(child.GetLogTree(depth + 1));
+            }
+            return root;
+        }
+    }
+
+    /// <summary>
+    /// Represents a node in the log tree.
+    /// </summary>
+    public class LogNode
```

Comment:
Could you move this to a different file? Maybe also put both files in a Logging folder?